front/components/database/DatabasePicker.tsx

```diff
@@ -0,0 +1,115 @@
+import { Menu } from "@headlessui/react";
+import { ChevronDownIcon } from "@heroicons/react/20/solid";
+
+import { CoreAPIDatabase } from "@app/lib/core_api";
+import { useDatabases } from "@app/lib/swr";
+import { classNames } from "@app/lib/utils";
+import { WorkspaceType } from "@app/types/user";
+
+export default function DatabasePicker({
+  owner,
+  dataSource,
+  currentDatabaseId,
+  readOnly,
+  onDatabaseUpdate,
+}: {
+  owner: WorkspaceType;
+  dataSource: {
+    workspace_id: string;
+    data_source_id: string;
+  };
+  currentDatabaseId?: string;
+  readOnly: boolean;
+  onDatabaseUpdate: (database: CoreAPIDatabase) => void;
+}) {
+  void owner;
+  void dataSource;
+
+  const { databases } = useDatabases({
+    workspaceId: dataSource.workspace_id,
+    dataSourceName: dataSource.data_source_id,
+    offset: 0,
+    // TODO(@fontanierh):
+    // will break if we have more than 100 databases (but UI wouldn't work anyway)
+    limit: 100,
+  });
+
+  const currentDatabase = currentDatabaseId
+    ? databases.find((db) => db.database_id === currentDatabaseId)
+    : null;
+
+  return (
+    <div className="flex items-center">
+      <div className="flex items-center">
+        {readOnly ? (
+          currentDatabase ? (
+            <div className="text-sm font-bold text-action-500">
+              {currentDatabase.name}
+            </div>
+          ) : (
+            "No Database"
+          )
+        ) : (
+          <Menu as="div" className="relative inline-block text-left">
+            <div>
+              <Menu.Button
+                className={classNames(
+                  "inline-flex items-center rounded-md py-1 text-sm font-normal text-gray-700",
+                  currentDatabase ? "px-0" : "border px-3",
+                  readOnly
+                    ? "border-white text-gray-300"
+                    : "border-orange-400 text-gray-700",
+                  "focus:outline-none focus:ring-0"
+                )}
+              >
+                {currentDatabase ? (
+                  <>
+                    <div className="mr-1 text-sm font-bold text-action-500">
+                      {currentDatabase.name}
+                    </div>
+                    <ChevronDownIcon className="mt-0.5 h-4 w-4 hover:text-gray-700" />
+                  </>
+                ) : databases && databases.length > 0 ? (
+                  "Select Database"
+                ) : (
+                  "There are no databases in this data source."
```

Comment:
THat's button-ish UI right? should this be "No Databases" ?

Reply:
It is button-ish indeed, will change 👍